backend/tests/test_integration_health.py

```
@@ -73,3 +83,247 @@ def test_health_sources_sends_degraded_notifications(self, client):
+                args=None,
+                exc_info=None,
+            )
+            record.request_id = "-"  # type: ignore[attr-defined]
+            # Drain whatever the worker may have pulled before stop() took
+            # effect, so the dropped-count assertion below measures our
+            # overflow delta only.
+            while True:
+                try:
+                    handler._q.get_nowait()
+                except Exception:
+                    break
```

Ruff (0.15.13)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)